oak_crypto/src/encryptor.rs

```diff
@@ -146,22 +146,26 @@ impl ClientEncryptor {
 /// Encryptor object for decrypting client requests that are received by the server and encrypting
 /// server responses that will be sent back to the client. Each Encryptor object corresponds to a
 /// single crypto session between the client and the server.
-/// Encryptor state is initialized after receiving an initial request message containing client's
-/// encapsulated public key.
 ///
 /// Sequence numbers for requests and responses are incremented separately, meaning that there could
 /// be multiple responses per request and multiple requests per response.
 pub struct ServerEncryptor {
-    recipient_context_generator: Arc<dyn RecipientContextGenerator>,
-    recipient_context: Option<RecipientContext>,
+    recipient_context: RecipientContext,
 }
 
 impl ServerEncryptor {
-    pub fn new(recipient_context_generator: Arc<dyn RecipientContextGenerator>) -> Self {
-        Self {
-            recipient_context_generator,
-            recipient_context: None,
-        }
+    pub fn create(
+        serialized_encapsulated_public_key: &[u8],
+        recipient_context_generator: Arc<dyn RecipientContextGenerator>,
+    ) -> anyhow::Result<Self> {
+        let recipient_context = recipient_context_generator
+            .generate_recipient_context(serialized_encapsulated_public_key)
+            .context("couldn't generate recipient crypto context")?;
+        Ok(Self::new(recipient_context))
+    }
```

Comment:
I'm trying to hide this logic away from the caller here, since the app develper already needs to know to extract the public key from the encrypted reqyest. Knowing about the context generation may be too much.

Reply:
And `RecipientContextGenerator` is basically implemented by the private key provider. So app developer needs to give the `create` method the client public key and the server private key, which should be simple enough.